spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/BackgroundPreinitializer.java

```diff
@@ -59,14 +60,21 @@
 	 */
 	public static final String IGNORE_BACKGROUNDPREINITIALIZER_PROPERTY_NAME = "spring.backgroundpreinitializer.ignore";
 
+	/**
+	 * Whether this environment lives within a native image.
+	 */
+	private static final boolean IN_NATIVE_IMAGE = (System.getProperty("org.graalvm.nativeimage.imagecode") != null);
```

Comment:
Would Graal's code elimination work if we had a single `ENABLED` field that was initialised using both the value of `spring.backgroundpreinitializer.ignore` and the presence of `org.graalvm.nativeimage.imagecode`?

Reply:
We could perhaps include the available processors check too:

```java
private static final boolean ENABLED;

static {
	ENABLED = System.getProperty("org.graalvm.nativeimage.imagecode") == null
		&& !Boolean.getBoolean(IGNORE_BACKGROUNDPREINITIALIZER_PROPERTY_NAME)
		&& Runtime.getRuntime().availableProcessors() > 1;
}
```

Reply:
Good point for the unique `ENABLED` flag, but I think we should not include the `Runtime.getRuntime().availableProcessors() > 1` check because the class will be initialized at build time on native image, so you will have the available processors for the build machine, not the runtime one.

Reply:
Oh but we don't care given the previous native image check so I guess we can do that.